helix-core/src/main/java/org/apache/helix/GroupCommit.java

```diff
@@ -139,14 +142,22 @@ public boolean commit(BaseDataAccessor<ZNRecord> accessor, int options, String k
           success = false;
           while (++retry <= MAX_RETRY && !success) {
             if (removeIfEmpty && merged.getMapFields().isEmpty()) {
-              success = accessor.remove(mergedKey, options);
+              try {
+                success = accessor.remove(mergedKey, options);
+              } catch (Exception e) {
+                success = false;
```

Comment:
Let us log this exception here.

Reply:
if you look at the next line. It's already logged there.

Reply:
The thing is that we did not log out the exception. At the time of investigation of production issue, you always want to know what is the exception thrown.

Current code actually "eat" the exception from IO (zk access). This is normally not a good idea.

how about something like this:

```
 try {
                success = accessor.remove(mergedKey, options);
  } catch (Exception e) {
                LOG.error (" failed to remove {} from zk, retry it", mergedkey, e);
                return false;
  }

 LOG.info( "removed {}", mergedkey);
```